src/mqtt/qmqtt_client_p.cpp

```diff
@@ -292,11 +303,15 @@ void QMQTT::ClientPrivate::startKeepAlive()
 {
     _timer.setInterval(_keepAlive*1000);
     _timer.start();
+    // The MQTT specification does not mention a timeout value in this case, so we use 10% of the
+    // keep alive interval.
+    _pingResponseTimer.setInterval(_keepAlive*100);
```

Comment:
I came across that problem too. We should not send ping requests if keep alive interval is set to 0. I think we should disable the keep alive timer if this happens.

About sending the ping request without checking for sent message: that is true. I didn't want to change this behavior because it is already present. I guess it is possible to change the behavior in the keep alive timer. Make it a single shot, and start it each time a message (including ping request) is sent to the server.

 @mwallnoefer / @KonstantinRitt : what do you think about this?

Reply:
you could check if _pingResponseTimer is active

Reply:
@KonstantinRitt Why would you do that? Given the class's logic, it should be stopped already.